Add info about community drivers to the docs

dacort opened the pull request:
Add information to our docs listing out existing community drivers. Fixes #10304

Also link to mailing list to subscribe to notifications for driver authors.

dacort committed:
Add info about community drivers to the docs

salsakran commented on docs/developers-guide-drivers.md:
```diff
@@ -0,0 +1,31 @@
+There are already a number of existing drivers out there. While we don't specifically endorse any of these drivers, we thought it would be helpful to have a central place to list them.
+
+## Community Database Drivers
+
```
I feel like this should be somewhere on www.metabase.com

This is pretty buried. For drivers that we think are high quality we should drive people towards them.

dacort replied:
Hm I'd say under maybe the "Open source and constantly improving" or "Works with" section in that case? 

One question, though, is what we define as high quality. @camsaul had some thoughts about this in his original PR #9087 (See `A Metabase plugins store`).

I think at the moment "high quality" is equivalent to "updated frequently" and some sort of validation that the driver works as intended.

camsaul replied:
If we wanted to dedicate the time to it we could make it easy to run tests against 3rd-party drivers or set up something like a Metabase plugins store. Since the plugins are loaded dynamically it would be entirely possible to download them and load them up without restarting the instance. So that's something to consider in the future. It's just a matter of prioritizing features, and it would take a while to implement either one

mazameli replied:
If these drivers go on the homepage, I think the "works with" section is the right place. 

I think I'd just list the names of the drivers in the main list, like "Athena." Each of the official drivers could possibly link to the doc page for how to connect to that DB, and all of the high quality community drivers that are listed could all go to the same interstitial page talking about community drivers, with links to corresponding repo pages. I.e., the "works with" section should IMO be more focused on answering the user question "can I use Metabase with the data I have?" rather than doing work to explain the distinction between drivers we've built vs. the community has built — that can be clarified as needed after a click.

dacort replied:
Cool, sounds good @mazameli 

And @camsaul I agree with you on that it would be nice to have a base set of tests we can run against 3rd-party drivers to validate different sets of functionality. I can think about how to structure that a bit more as I just went through adding CI testing to the Athena driver.

salsakran replied:
what was in my head with "High Quality"  was
1) Full support for the driver API (i.e. some community drivers have only supported native queries)
2) Passes all driver tests
3) Is actively being maintained by someone who commits to keeping them mostly caught up to the latest version.

dacort replied:
Sounds good - will need to figure out a way to validate #1 and #2 but #3 is pretty easy to keep an eye on. And one of the reasons I added "last release date" to the table. Will rework this a little bit.

dacort replied:
OK - I added a PR for the website that adds a few drivers that are actively developed and links back to this documentation. I'll open a feature request to make testing/validation easier for third-party drivers.

I also tweaked this documentation a little bit so it's a little bit more friendly for somebody clicking on the home page and wondering "hey do I use that thing?".

dacort replied:
Added my thoughts for tests to #9348

dacort committed:
Update driver docs
- opened relevant PR for website to add some active drivers there that link to this documentation

mazameli commented on docs/developers-guide-drivers.md:
```diff
@@ -0,0 +1,39 @@
+
+If the driver you're looking for isn't available, take a look at the [Writing a Driver](https://github.com/metabase/metabase/wiki/Writing-a-Driver) wiki page. It's still a work in progress, but should give you a good start. A few things to keep in mind:
+
+- If your database has a JDBC driver, you'll be able to make use some common classes that already exist in Metabase.
```
```suggestion
- If your database has a JDBC driver, you'll be able to make use of some common classes that already exist in Metabase.
```

mazameli commented on docs/developers-guide-drivers.md:
```diff
@@ -0,0 +1,39 @@
+# Metabase Driver Development Guide
```
I think this title should be changed to something like "Community-built Database Drivers" for two reasons:
1. It'll make more sense as a heading if you've arrived here from clicking on the name of a community driver on the homepage.
2. This page has content both about using existing drivers as well as developing them; it's not strictly about how to develop.

dacort replied:
*nod* Makes sense. I had originally linked to this page under the `For Developers` section in the base of the docs. Do you think that's still relevant or should we tweak that as well>

mazameli replied:
I think that's okay, because now there are three entry points to this page:
1. the homepage for folks who are trying to see if they can use MB with their DB
2. the "managing databases" admin doc page which links to this page for folks who are looking for a driver not listed in the officially supported DBs list
3. the link under "for developers" as you mentioned for folks who want to try writing a driver

mazameli commented on docs/developers-guide-drivers.md:
```diff
@@ -0,0 +1,39 @@
+
+Want to build your own driver? Take a look at the [driver development](#driver-development) section below.
+
+## Community Database Drivers
```
If the main title is changed as I suggested, this one should change to be more different from the main title, like "How to use a community-built driver."

dacort committed:
Address review feedback

dacort commented:
@mazameli Thanks for the review - just pushed a new revision with the changes requested.

mazameli commented on docs/developers-guide-drivers.md:
```diff
@@ -1,12 +1,12 @@
 Want to build your own driver? Take a look at the [driver development](#driver-development) section below.
 
-## Community Database Drivers
+## How to use a community-built driver
```
All of your other headings and subheadings use Title Case — pick a side, man! 😂 

FWIW, my personal preference is for `Sentence case` for headings and subheadings because I think it looks less formal. Our existing docs aren't perfectly consistent, but they primarily tend to use sentence case (because I wrote a whole lot of them :b ). 

It's probably time for me to bang out a draft of a docs/copy-writing style guide to collaborate on with you and @kdoh (he might have even started a draft like a year ago).

dacort replied:
🤣 I just want to give every heading equal opportunity! 

I'll go through and make them more consistent. 🙏

mazameli reviewed (approved):
This looks good to go minus the nit-picky detail of title vs. sentence case for headings.

dacort committed:
Maintain consistent headings :)
